lib/ansible/modules/cloud/azure/azure_rm_sql_server.py

```diff
@@ -0,0 +1,292 @@
+            - Resource location.
+        required: True
+    tags:
+        description:
+            - Resource tags.
+        required: False
+    identity:
+        description:
+            - The Azure Active Directory identity of the server.
+        required: False
+        suboptions:
+            type:
+                description:
+                    - "The identity type. Set this to 'SystemAssigned' in order to automatically create and assign an Azure Active Directory principal for th
+                       e resource. Possible values include: 'SystemAssigned'"
+                required: False
+    administrator_login:
+        description:
+            - Administrator username for the server. Once created it cannot be changed.
+        required: False
+    administrator_login_password:
+        description:
+            - The administrator login password (required for server creation).
+        required: False
+    version:
+        description:
+            - The version of the server.
+        required: False
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+      - name: Create (or update) Servers
+        azure_rm_sql_server:
+          resource_group: "{{ resource_group }}"
+          name: zims-server
+          location: westus
+          tags: "{{ tags }}"
+          identity:
+            type: "{{ type }}"
+          administrator_login: mylogin
+          administrator_login_password: Testpasswordxyz12!
+          version: "{{ version }}"
+'''
+
+RETURN = '''
+state:
+    description: Current state of Server
+    returned: always
+    type: dict
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.sql import SqlManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMServers(AzureRMModuleBase):
+    """Configuration class for an Azure RM Server resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str',
+                required=True
+            ),
+            tags=dict(
+                type='dict',
+                required=False
+            ),
+            identity=dict(
+                type='dict',
+                required=False
+            ),
+            administrator_login=dict(
+                type='str',
+                required=False
+            ),
+            administrator_login_password=dict(
+                type='str',
+                required=False
```

Comment:
Add another entry `no_log=True` to stop Ansible from logging this value, this should be set on any secret parameters (like passwords) to keep it a secret.